app/models/container_group.rb

```diff
@@ -37,6 +38,25 @@ def ready_condition_status
     ready_condition.try(:status) || 'None'
   end
 
+  def containers_states_summary
+    containers = self.containers.to_a
+    {
+      :waiting    => containers_state_count('waiting', containers),
+      :running    => containers_state_count('running', containers),
+      :terminated => containers_state_count('terminated', containers)
```

Comment:
@simon3z I checked the query log, although the array is iterated 3 times it seems that only one SQL query is generated. It can be changed to one iteration over the array of course, but as far as I can tell it does not affect the number of queries.